### ISSUE TITLE
Afterburner breaks serialization of ObjectMapper

### ISSUE THREAD

**Issue by zman0900** [FasterXML/jackson-modules-base#97]
JsonMapper and its parent class ObjectMapper are both supposed to be Serializable, but when AfterburnerModule is added, serialization is no longer possible.

For example, this code:
```Java
public class SerTest {
    public static void main(final String... args) throws Exception {
        final JsonMapper x = JsonMapper.builder()
                .addModule(new AfterburnerModule())
                .build();

        final ByteArrayOutputStream out = new ByteArrayOutputStream();
        new ObjectOutputStream(out)
                .writeObject(x);

        final ObjectInputStream input = new ObjectInputStream(new ByteArrayInputStream(out.toByteArray()));
        final JsonMapper y = (JsonMapper) input.readObject();
    }
}
```

Results in exception:

```
Exception in thread "main" java.io.NotSerializableException: com.fasterxml.jackson.module.afterburner.deser.DeserializerModifier
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1184)
	at java.io.ObjectOutputStream.writeArray(ObjectOutputStream.java:1378)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1174)
	at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1548)
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1509)
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1432)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1178)
	at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1548)
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1509)
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1432)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1178)
	at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1548)
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1509)
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1432)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1178)
	at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1548)
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1509)
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1432)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1178)
	at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:348)
	at SerTest.main(SerTest.java:13)
```

Without adding the AfterburnerModule to JsonMapper, this works as expected.

Jackson version = 2.11.0

**Comment by cowtowncoder** [comment]
Thank you for reporting this: yes, modules should be serializable (there is one known exception for 2.x, Java 8 date/time module; but those are exceptions to the rule and Afterburner not known as one).

One quick question: do you know if this is a regression (used to work), or just observed now?

**Comment by cowtowncoder** [comment]
Ok. So, was able to make serializability work in 2.11(.1) as long as `ObjectMapper` serialized BEFORE any use of Afterburner (that is, readValue, writeValue that triggers class generation).
This is likely fundamental limitation for 2.x: 3.0 will fix the problem as mapper serialization includes much smaller set of data (basically only configuration), but with 2.x there are limits.

**Comment by zman0900** [comment]
I've never tried it in previous versions. I've worked around it for now by just using a static variable. I was trying to have Spring create and configure the JsonMapper, which would then be used in a map function for Spark, but those must be serializable.

**Comment by cowtowncoder** [comment]
@zman0900 Makes sense. Serializing `ObjectMapper` is often best avoided but intent is for JDK serialization to work for mapper, modules, as sometimes it is still needed.

As per my notes this will be fixed in 2.11.1, as long as pattern of config-only-before-serialization/deserialization is followed.
